src/realm/exceptions.hpp

```diff
@@ -406,6 +406,20 @@ struct InvalidQueryArgError : InvalidArgument {
 };
 
 } // namespace query_parser
+
+namespace sync {
+
+// Exception thrown when the request/websocket URL is malformed
+class BadServerUrl : public Exception {
```

Comment:
nit, but shouldn't this just be `BadUrl`?

Reply:
I just moved it as-is from `sync_session.hpp`. I can rename it, if you prefer.